### PR TITLE
feat: add universe support to googleapis libraries

### PR DESCRIPTION
**Pull request by alexander-fenster**
Adding a global option `universeDomain` (can also be spelled as `universe_domain`). Example usage with `google-api-nodejs-client` (for Cloud KMS):

```ts
import * as cloudkms from '@googleapis/cloudkms';

const googleAuth = new auth.GoogleAuth({
  scopes: [
    'https://www.googleapis.com/auth/cloud-platform',
    'https://www.googleapis.com/auth/cloudkms',
  ],
});
const kms = cloudkms.cloudkms({
  auth: googleAuth,
  version: 'v1',
  universeDomain: 'example.com',  // use your Trusted Partner Cloud universe domain
});
const project = await googleAuth.getProjectId();
const location = 'some-existing-location-such-as-us-central1';
const result = await kms.projects.locations.keyRings.list({
  parent: `projects/${project}/locations/${location}`,
});
console.log(result.data);
```

The universe domain can be also configured with `GOOGLE_CLOUD_UNIVERSE_DOMAIN` environment variable.

Note that for services which have modern libraries published to `@google-cloud/` NPM organization (e.g. `@google-cloud/kms`), we recommend using them and not `@googleapis` libraries. Still adding this feature to `@googleapis` for completeness.

### CODE REVIEW

File: src/apirequest.ts [hunk 1]
```diff
@@ -160,7 +160,11 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
 
   // Parse urls
   if (options.url) {
-    options.url = urlTemplate.parse(options.url).expand(params);
+    let url = options.url;
```
Comment (alexander-fenster) [inline src/apirequest.ts:163]: This is an unrelated fix: without this, the existing code does not compile on the new TypeScript version.

File: package.json [hunk 2]
```diff
@@ -69,7 +69,7 @@
     "karma-mocha": "^2.0.0",
     "karma-remap-coverage": "^0.1.5",
     "karma-sourcemap-loader": "^0.4.0",
-    "karma-webpack": "^5.0.0",
+    "karma-webpack": "^4.0.0",
```
Comment (alexander-fenster) [inline package.json:72]: This is unrelated, `webpack` and `karma-webpack` v5 require Node 18 and it breaks tests. These are dev dependencies, we don't really care.

**Review by sofisl — APPROVED** [review]

**Review by viacheslav-rostovtsev — APPROVED** [review]
LGTM
